src/main/java/com/google/step/YTLounge/servlets/SingleVideoSearch.java

```diff
@@ -0,0 +1,170 @@
+    try {
+      youtubeService = getService();
+    } catch (GeneralSecurityException e) {
+      response.getWriter().println("");
+    }
+    if (youtubeService == null) {
+      return;
+    }
+    YouTube.Videos.List videoRequest = youtubeService.videos().list("snippet,contentDetails");
+    VideoListResponse videoResponse = videoRequest.setId(videoID).execute();
+    JsonObject jsonObject = new JsonParser().parse(gson.toJson(videoResponse)).getAsJsonObject();
+    JsonObject error = jsonObject.getAsJsonObject("error");
+    if (error != null) { // if video doesn't exist
+      return;
+    }
+
+    extractVideo(jsonObject.getAsJsonArray("items"), videoID);
+    response.getWriter().println(gson.toJson(videoResponse));
+  }
+
+  /**
+   * If a room wasn't found from the query, create a new room ID while respecting the current IDs in
+   * DataStore. Initializes necessary properties for the room.
+   *
+   * @return a string representing a room's identifier
+   */
+  private String generateRoomID() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity roomEntity = new Entity("room");
+    roomEntity.setProperty("members", new HashSet<Key>());
+    roomEntity.setProperty("nowPlaying", "");
+    roomEntity.setProperty("queue", new ArrayList<Key>());
+    roomEntity.setProperty("duration", 0);
+    roomEntity.setProperty("elapsedTime", 0);
+    roomEntity.setProperty("log", new ArrayList<Key>());
+    datastore.put(roomEntity);
+    return Long.toString(roomEntity.getKey().getId());
+  }
+
+  /**
+   * Iterates through the given items and locates specific values to create a new Video and upload
+   * the video to DataStore
+   */
+  private void extractVideo(JsonArray items, String videoID) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity videoEntity = new Entity("Video");
+    for (int i = 0; i < items.size(); i++) {
+      JsonObject snippet = items.get(i).getAsJsonObject().getAsJsonObject("snippet");
+      String thumbnailURL =
+          snippet.getAsJsonObject("thumbnails").getAsJsonObject("medium").get("url").toString();
+      String title = snippet.get("title").toString();
+      String duration =
+          items
+              .get(i)
+              .getAsJsonObject()
+              .getAsJsonObject("contentDetails")
+              .get("duration")
+              .toString();
+      String formattedVideoURL = "https://youtube.com/watch?v=" + videoID;
+      String channelName = snippet.get("channelTitle").toString();
+      String releaseDate = snippet.get("publishedAt").toString();
+      videoEntity.setProperty("title", title);
+      videoEntity.setProperty("thumbnailURL", thumbnailURL);
+      videoEntity.setProperty("videoURL", formattedVideoURL);
+      videoEntity.setProperty("videoID", videoID);
+      videoEntity.setProperty("duration", parseDuration(duration));
+      videoEntity.setProperty("channelName", channelName);
```

Comment:
when retrieving the channel name, does that also come with if the channel has the verified checkmark or the "official artist channel" mark (just curious)